src/Console/Output/ProcessOutputFactory.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output;
+
+final class ProcessOutputFactory
+{
+    public static function create(string $outputType, OutputContext $context): ProcessOutputInterface
```

Comment:
no static methods pls.

static is as evil as global.
good luck for whoever later to introduce proper DIC or mocking ProcessOutputFactory as dependency.

Reply:
Good luck with mocking current `FixCommand`'s dependencies too:

```php
public function __construct(ToolInfoInterface $toolInfo)
{
    parent::__construct();

    $this->eventDispatcher = new EventDispatcher();
    $this->errorsManager = new ErrorsManager();
    $this->stopwatch = new Stopwatch();
    $this->defaultConfig = new Config();
    $this->toolInfo = $toolInfo;
}
```

😉 

`ProgressOutputFactory` is used within `FixCommand` too, this is the same level of non-interoperability as current objects initialised in the constructor. What's the point in making it non-static and creating instance in the command's constructor, just to call one factory method? 🤷‍♂️ 

But yeah, if you really want, I can change it, even though I don't see much sense in it.

Reply:
> no static methods pls.
> 
> static is as evil as global.

quite a subjective opinion, there is nothing wrong with reasonable static method usage: https://verraes.net/2014/06/when-to-use-static-methods-in-php/

Reply:
if Foo is ugly, it's not a reason that newly introduce Bar should be ugly too.

Reply:
I'd advocate that stateless static function is much less ugly that using `new` in constructor.

Reply:
i decided to limit myself to ask for improvement in code introduced in this PR, not to ask to improve whole codebase ;)

Reply:
OK, so what's the conclusion? Remove `static` from method and initialise `ProgressOutputFactory` in the `FixCommand`'s constructor just like other dependencies? I would like to minimise risk of another round of review 😉

Reply:
yep.
and then ultimately [in one other PR, in 2030], stop doing it in constructor but inject it from DIC.